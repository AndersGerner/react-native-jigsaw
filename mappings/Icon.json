{"name":"Icon","tag":"Icon","description":"An icon","category":"basic","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Icon.png","supports_list_render":false,"props":{"name":{"label":"Name","description":"Name of the icon","type":"icon","value":null,"editable":true,"required":true},"color":{"label":"Color","description":"Color of the icon","type":"color","value":"strong","editable":true,"required":true},"size":{"label":"Size","description":"Width and height of the icon","type":"number","value":24,"min":1,"max":200,"step":1,"precision":0,"editable":true,"required":true}},"layout":{"width":24,"height":24}}