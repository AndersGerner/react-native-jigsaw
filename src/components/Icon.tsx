// @ts-nocheck
import { ViewProps, StyleProp, ImageStyle } from "react-native";

interface Props extends ViewProps {
  name: string | number | { uri: string };
  color?: string;
  size: number;
  style?: StyleProp<ImageStyle>;
}

const Icon: React.FC<Props> = ({ _name, _color, _size, _style }) => {
  return null;
};

export default Icon;
