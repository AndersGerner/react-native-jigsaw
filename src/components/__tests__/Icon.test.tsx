import * as React from "react";
import renderer from "react-test-renderer";
import Icon from "../Icon";

it("renders icon", () => {
  const tree = renderer.create(<Icon size={10} name="arrow" />).toJSON();

  expect(tree).toMatchSnapshot();
});
