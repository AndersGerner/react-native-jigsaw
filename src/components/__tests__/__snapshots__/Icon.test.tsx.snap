// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders icon 1`] = `null`;
